### ISSUE TITLE
Issue with default XMODIFIERS setting: keyboard not working in CalibreDRC/CalibreLVS

### ISSUE THREAD

**Issue by naatje80** [TigerVNC/tigervnc#974]
**Describe the bug**
We are testing TigerVNC as a replacement for a commercial alternative we are currently using. Last week we encountered a bug with Mentor CalibreDRC CalibreLVS: The keyboard input is not excepted in most input fields. The following topic seems to be related:
https://askubuntu.com/questions/973748/why-is-vnc-not-able-to-change-the-keyboard-layout-when-the-main-display-can
It pointed me in the right direction. I also tried TurboVNC, which did not seem to encounter the same bug. I copied the setting of the XMODIFIERS and QT_IM_MODULE (to ensure both are similar to the TurboVNC env.) and changed them to the TurboVNC defaults:
QT_IM_MODULE=xim
XMODIFIERS=@im=none
This solves the issue. Furthermore the above setup (at least for QT_IM_MODULE) also seems to be setup according to the way the ibus deamon has been started for Gnome Shell:
`my_account 24800 24580  0 Mar05 ?        00:00:25 /usr/bin/ibus-daemon -r --xim`

**To Reproduce**
Specific IC Design verification software (At the moment we can not find any other application that encounters similar issues, although googling on this issue seems that it could cause issue for other software as well). But In case someone is able to do test it:
1. Open Cadence Virtuoso
2. Open a Layout/Schematic
3. Open CalibreDRC/CalibreLVS/CalibreDRC
4. Try to type something in one of setup input fields of the Mentor Calibre LVS/DRC Window. Most fields do not except keyboard input (copy/past with middle mouse button seems to work however).

**Expected behavior**
One should be able to change values of the various input fields of the Mentor Calibre Windows.

**Client (please complete the following information):**
 - OS: Windows 10
 - VNC client: TurboVNC (Although TigerVNC 1.10.1 is also encountering the same issue; it is server setup related)
 - VNC client version: 2.2.4
 - Client downloaded from: TigerVNC: Manually compiled, sources downloaded from GitHub, TurboVNC client form: https://www.turbovnc.org/

**Server (please complete the following information):**
 - OS: Scientific Linux 7.6/7.7
 - VNC server: TigerVNC
 - VNC server version: 1.10.1
 - Server downloaded from: Github, Manually compiled
 - Server was started using: tigervnc -listen tcp

**Comment by CendioOssman** [comment]
This will be difficult for us to address if we cannot reproduce it in some easily available software. :/

Perhaps it has something to do with the startup though and #838 will resolve it?

Can you reproduce the issue with other desktop environments?

**Comment by naatje80** [comment]
@CendioOssman Thank you for the update, I understand that it is difficult to test from your side. I was also just wondering if the default settings are correct, and why TurboVNC has different settings with the same desktop envorment.  I will check with an alternative environment and do some further testing. I will keep you updated....

**Comment by naatje80** [comment]
@CendioOssman Thank you! I indeed resolved the issue by removing our manually created $HOME/.vnc/xstatup file and changing the $HOME/.vnc/config instead (by changing the desktop value to gnome-classic). From what I understand from #838, the xstartup file is no longer supported. But I've build this specific branch, and still noticed that the xstartup file is called. Is this something that will change in the near future, or does it mean we simply no longer are able to change it (because the default should be used)?

**Comment by CendioOssman** [comment]
The `desktop` setting only controls the title of the session. It's `session` that controls what is started. But that setting only has effect on #838, at which point `xstartup` is no longer used. So if `xstartup` was still used in your case then I'd say that you didn't get that branch up and running and must still be using the current stuff.

**Comment by naatje80** [comment]
@CendioOssman Thank you for you clarification. In that case I indeed must have made a mistake during checkout of the correct branch. But at least it helped me to find the source of the issue. The xstatup we generate seems to be causing the incorrect XMODIFIERS setting. I will check the de default xstartup that is generated by tigervnc to investigate what we should add to our environment.

**Comment by naatje80** [comment]
Today we found out the issue was not related to xstartup files at all: at time of testing we were using a mix. of EL 7.6 and EL 7.7 servers. But now all servers have been updated, we noted that it was caused by a change in the ibus packages, specifically this change:
`Resolves: #1671187 - ibus.conf does not set QT_IM_MODULE with qt5 ibus module`
For now we resolved this issue by manually changing the XMODIFIERS variable before starting the actual application. 

I also read more applications could be affected, so possible a change will be implemented in the near future.

@CendioOssman Thank you again for your help!
